tests/pci_passthrough/conftest.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import pytest
+
+from lib.common import safe_split
+
+@pytest.fixture(scope="session")
+def enabled_pci_uuid(host_at_least_8_3):
+    host = host_at_least_8_3
+    pci_uuids = safe_split(host.xe("pci-list", {"host-uuid": host.uuid}, minimal=True), ',')
+
+    pci_uuid = None
+    for uuid in pci_uuids:
+        dom0_access = host.xe("pci-get-dom0-access-status", {"uuid": uuid})
+        if dom0_access == "enabled":
+            pci_uuid = uuid
+            break
+
+    if pci_uuid is None:
+        pytest.skip("This test requires a PCI to hide from dom0")
+
+    yield pci_uuid
+
+    # Put PCI back in initial state
+    if host.xe("pci-get-dom0-access-status", {"uuid": pci_uuid}) != "enabled":
+        host.xe("pci-enable-dom0-access", {"uuid": pci_uuid})
+        if host.xe("pci-get-dom0-access-status", {"uuid": pci_uuid}) != "enabled":
+            host.reboot(verify=True)
```

Comment:
I'm not very comfortable with a fixture causing a reboot at the end of the tests, but I don't really have a better suggestion at the moment :)

Reply:
I try to reboot the least possible (ie in case of bad tests execution) in order to leave the host in the same state it began.